android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreRSAECB.java

```diff
@@ -0,0 +1,339 @@
+        }
+
+        if (mActivity == null) {
+            throw new Exception("mActivity is null (make sure to call setCurrentActivity)");
+        }
+
+        mBiometricPromptCompatCancellationSignal = new CancellationSignal();
+
+        mBiometricPromptCompat = new BiometricPromptCompat.Builder(mActivity)
+            .setTitle("Authentication required")
+            .setSubtitle("Please use biometric authentication to unlock the app")
+            .build();
+
+        mBiometricPromptCompat.authenticate(mBiometricPromptCompatCancellationSignal, this);
+    }
+
+    @Override
+    public String getCipherStorageName() {
+        return CIPHER_STORAGE_NAME;
+    }
+
+    @Override
+    public boolean getCipherBiometrySupported() {
+        return true;
+    }
+
+    @Override
+    public int getMinSupportedApiLevel() {
+        return Build.VERSION_CODES.M;
+    }
+
+    @Override
+    public EncryptionResult encrypt(@NonNull String service, @NonNull String username, @NonNull String password) throws CryptoFailedException {
+        service = getDefaultServiceIfEmpty(service);
+
+        try {
+            KeyStore keyStore = getKeyStoreAndLoad();
+
+            if (!keyStore.containsAlias(service)) {
+                generateKeyAndStoreUnderAlias(service);
+            }
+
+            KeyFactory keyFactory = KeyFactory.getInstance(ENCRYPTION_ALGORITHM);
+            PublicKey publicKey = keyStore.getCertificate(service).getPublicKey();
+            KeySpec spec = new X509EncodedKeySpec(publicKey.getEncoded());
+            Key key = keyFactory.generatePublic(spec);
+
+            byte[] encryptedUsername = encryptString(key, service, username);
+            byte[] encryptedPassword = encryptString(key, service, password);
+
+            return new EncryptionResult(encryptedUsername, encryptedPassword, this);
+        } catch (NoSuchAlgorithmException | InvalidAlgorithmParameterException | NoSuchProviderException e) {
+            throw new CryptoFailedException("Could not encrypt data for service " + service, e);
+        } catch (KeyStoreException | KeyStoreAccessException e) {
+            throw new CryptoFailedException("Could not access Keystore for service " + service, e);
+        } catch (Exception e) {
+            throw new CryptoFailedException("Unknown error: " + e.getMessage(), e);
+        }
+    }
+
+    private void generateKeyAndStoreUnderAlias(@NonNull String service) throws NoSuchAlgorithmException, NoSuchProviderException, InvalidAlgorithmParameterException {
+        AlgorithmParameterSpec spec = new KeyGenParameterSpec.Builder(
+                service,
+                KeyProperties.PURPOSE_DECRYPT | KeyProperties.PURPOSE_ENCRYPT)
+                .setBlockModes(ENCRYPTION_BLOCK_MODE)
+                .setEncryptionPaddings(ENCRYPTION_PADDING)
+                .setRandomizedEncryptionRequired(true)
+                .setUserAuthenticationRequired(true)
+                .setUserAuthenticationValidityDurationSeconds(-1)
```

Comment:
This needs to be at-least 2 seconds. Otherwise the decryption will always fail.

To work around the issue we need to pass in a CryptoObject to the `.authenticate` call (as mentioned here https://github.com/oblador/react-native-keychain/issues/116#issuecomment-420657678) but I could not figure out myself how to do it. I can't seem to be able to figure out how to pass the Cipher to CryptoObject since it needs to be initialized, and to initialize it user needs to be authenticated... Hopefully @cladjules will figure that one out.

Reply:
That's why we need to get the crypto object out of encryptBytes,
getting the instance of the Cipher should be fine.

Then, try to call cipher.init, that should throw the UserNotAuthenticatedException (or not).

I will give it a go on that PR.

Reply:
Optional ... setIsStrongBoxBacked